frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -162,6 +168,71 @@ const osUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
   if (!isWindows && windowsTools) {
     dispatch(vmWizardInternalActions[InternalActionType.RemoveStorage](id, windowsTools.id));
   }
+
+  const iCommonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+  const iTemplate = iCommonTemplates && iGetOSTemplate(iCommonTemplates, os);
+  const pvcName =
+    iTemplate && iGetAnnotations(iTemplate).get(`${TEMPLATE_DATAVOLUME_ANNOTATION}/${os}`);
+
+  const iBaseImages = iGetLoadedCommonData(state, id, VMWizardProps.openshiftCNVBaseImages);
+  const iBaseImage =
+    pvcName &&
+    iBaseImages &&
+    iBaseImages
+      .valueSeq()
+      .filter((iPVC) => iGetName(iPVC) === pvcName)
+      .first();
+
+  dispatch(
+    vmWizardInternalActions[InternalActionType.UpdateVmSettingsField](
+      id,
+      VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE,
+      {
+        isHidden: asHidden(!iBaseImage, VMSettingsField.OPERATING_SYSTEM),
+        value: !!iBaseImage,
+      },
+    ),
+  );
+};
+
+const cloneCommonBaseDiskImageUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
+  const state = getState();
+  if (
+    !hasVMSettingsValueChanged(prevState, state, id, VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE)
+  ) {
+    return;
+  }
+
+  const userTemplate = iGetVmSettingValue(state, id, VMSettingsField.USER_TEMPLATE);
+  const cloneCommonBaseDiskImage = iGetVmSettingValue(
+    state,
+    id,
+    VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE,
+  );
+
+  dispatch(
+    vmWizardInternalActions[InternalActionType.UpdateVmSettings](id, {
+      [VMSettingsField.PROVISION_SOURCE_TYPE]: {
+        isHidden: asHidden(cloneCommonBaseDiskImage, VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE),
+      },
+      [VMSettingsField.CONTAINER_IMAGE]: {
+        isHidden: asHidden(cloneCommonBaseDiskImage, VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE),
+      },
+      [VMSettingsField.IMAGE_URL]: {
+        isHidden: asHidden(cloneCommonBaseDiskImage, VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE),
+      },
+    }),
+  );
+
+  if (!userTemplate) {
```

Comment:
this better be unconditional as it can lead to weird bugs when switching templates

Reply:
ah I just read. https://github.com/openshift/console/pull/5977#discussion_r460493570

in that case please use the same dispatch and write and resolve value accordingly (undefined means no update will occur) and then please write a comment why are we passing undefined in a case userTemplate is not defined

Reply:
done